circuit/src/storage_proof.rs

```diff
@@ -118,11 +149,19 @@ impl CircuitFragment for StorageProof {
             prev_hash = hashes[i];
         }
 
+        // Now do leaf hash TODO: how?
+        let leaf_hash = builder.hash_n_to_hash_no_pad(leaf_inputs.clone());
+        for y in 0..4 {
+            let leaf_diff = builder.sub(leaf_hash.elements[y], prev_hash.elements[y]);
```

Comment:
prev_hash here is the last target in the array, which will typically be empty, which is not what we want. Can I put a conditional based on the known (private) length of the proof?